bots/tests-policy

```diff
@@ -0,0 +1,360 @@
+            (name, delim, directive) = line.partition("#")
+            (name, delim, directive) = name.partition("duration")
+            name = name.strip()
+            return name
+    return ""
+
+# -----------------------------------------------------------------------------
+# Retry policy
+
+def checkRetry(trace):
+    # We check for persistent but test harness or framework specific
+    # failures that otherwise cause flakiness and false positives.
+    #
+    # The things we check here must:
+    #  * have no impact on users of Cockpit in the real world
+    #  * be things we tried to resolve in other ways. This is a last resort
+    #
+
+    trace = normalize_traceback(trace)
+
+    # HACK: An issue in phantomjs and QtWebkit
+    # http://stackoverflow.com/questions/35337304/qnetworkreply-network-access-is-disabled-in-qwebview
+    # https://github.com/owncloud/client/issues/3600
+    # https://github.com/ariya/phantomjs/issues/14789
+    if "PhantomJS or driver broken" in trace:
+        return True
+
+    # HACK: A race issue in phantomjs that happens randomly
+    # https://github.com/ariya/phantomjs/issues/12750
+    if "Resource Error: Operation canceled" in trace:
+        return True
+
+    # HACK: Interacting with sshd during boot is not always predictable
+    # We're using an implementation detail of the server as our "way in" for testing.
+    # This often has to do with sshd being restarted for some reason
+    if "SSH master process exited with code: 255" in trace:
+        return True
+
+    # HACK: Intermittently the new libvirt machine won't get an IP address
+    # or SSH will completely fail to start. We've tried various approaches
+    # to minimize this, but it happens every 100,000 tests or so
+    if "Failure: Unable to reach machine " in trace:
+        return True
+
+    # HACK: For when the verify machine runs out of available processes
+    # We should retry this test process
+    if "self.pid = os.fork()\nOSError: [Errno 11] Resource temporarily unavailable" in trace:
+        return True
+
+    return False
+
+
+# -----------------------------------------------------------------------------
+# Known Issue Matching and Filing
+
+def normalize_traceback(trace):
+    # All file paths converted to basename
+    return re.sub(r'File "[^"]*/([^/"]+)"', 'File "\\1"', trace.strip())
+
+def list_directories(dirs):
+    result = [ ];
+    for d in dirs:
+        for f in os.listdir(d):
+            result.append(os.path.join(d, f))
+    return result
+
+def check_known_issue(api, trace, image):
+    directories =  [ ]
+    image_naughty = os.path.join(BOTS, "naughty", image)
+    if os.path.exists(image_naughty):
+        directories.append(image_naughty)
+    trace = normalize_traceback(trace)
+    number = 0
+    for naughty in list_directories(directories):
+        (prefix, unused, name) = os.path.basename(naughty).partition("-")
+        try:
+            n = int(prefix)
+        except:
+            continue
```

Comment:
Blanket except is something to avoid (`ValueError`). Also, I'd really not silently skip this, as having a file without an issue number prefix in `bots/naughty/` is almost surely a human oversight, and should cause a test failure.

Reply:
We could try and drop it, but this is probably material for a follow up, so we don't change too many things. This is just a renamed file after all.